### PR TITLE
Update imports from sdk-backend-spi to sdk-model

### PR DESCRIPTION
**Pull request by kandl**
Update imports of all interfaces, types and typeguards that were moved from sdk-backend-spi to sdk-model across the whole UI.SDK codebase.

JIRA: RAIL-4109

---

Supported PR commands:

| Command                  | Description             |
| ------------------------ | ----------------------- |
| `ok to test`             | Re-run standard checks  |
| `extended test`          | BackstopJS tests        |
| `extended check sonar`   | SonarQube tests         |
| `extended check cypress` | Cypress E2E tests       |
| `extended check plugins` | Dashboard plugins tests |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `check-extended-cypress` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

File: libs/api-client-tiger/api/api-client-tiger.api.md [hunk 1]
```diff
@@ -7,10 +7,10 @@
 import { AxiosInstance } from 'axios';
 import { AxiosPromise } from 'axios';
 import { IBucket } from '@gooddata/sdk-model';
-import { IDashboardDateFilterConfig } from '@gooddata/sdk-backend-spi';
-import { IDashboardLayout } from '@gooddata/sdk-backend-spi';
+import { IDashboardDateFilterConfig } from '@gooddata/sdk-model';
```
Comment (no23reason) [inline libs/api-client-tiger/api/api-client-tiger.api.md:10]: Maybe we can now cut the dependency between api-client-tiger and sdk-backend-spi?

File: libs/sdk-ui-dashboard/api/sdk-ui-dashboard.api.md [hunk 2]
```diff
@@ -20,105 +20,98 @@ import { DateFilterType } from '@gooddata/sdk-backend-spi';
...
 import { Dispatch } from '@reduxjs/toolkit';
-import { DrillDefinition } from '@gooddata/sdk-backend-spi';
+import { DrillDefinition } from '@gooddata/sdk-model';
 import { EntityId } from '@reduxjs/toolkit';
 import { EntityState } from '@reduxjs/toolkit';
 import { ExplicitDrill } from '@gooddata/sdk-ui';
-import { FilterContextItem } from '@gooddata/sdk-backend-spi';
-import { FilterContextItem as FilterContextItem_2 } from '@gooddata/sdk-model';
+import { FilterContextItem } from '@gooddata/sdk-model';
+import { FilterContextItem as FilterContextItem_2 } from '@gooddata/sdk-backend-spi';
```
Comment (no23reason) [inline libs/sdk-ui-dashboard/api/sdk-ui-dashboard.api.md:28]: Seems there is some spi import left behind.

File: libs/sdk-ui-filters/api/sdk-ui-filters.api.md [hunk 3]
```diff
@@ -5,15 +5,15 @@
...
-import { IAbsoluteDateFilterPreset } from '@gooddata/sdk-backend-spi';
-import { IAllTimeDateFilterOption } from '@gooddata/sdk-backend-spi';
+import { DashboardDateFilterConfigMode } from '@gooddata/sdk-model';
+import { DateFilterGranularity } from '@gooddata/sdk-model';
+import { DateString } from '@gooddata/sdk-model';
+import { IAbsoluteDateFilterForm } from '@gooddata/sdk-model';
+import { IAbsoluteDateFilterPreset } from '@gooddata/sdk-model';
+import { IAllTimeDateFilterOption } from '@gooddata/sdk-model';
 import { IAnalyticalBackend } from '@gooddata/sdk-backend-spi';
 import { IAttributeElement } from '@gooddata/sdk-backend-spi';
```
Comment (no23reason) [inline libs/sdk-ui-filters/api/sdk-ui-filters.api.md:15]: Seems there is some forgotten SPI import still.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/7494/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/9353/ : SUCCESS in 16m 33s

File: libs/sdk-ui/api/sdk-ui.api.md [hunk 4]
```diff
@@ -45,16 +45,16 @@ import { IPositiveAttributeFilter } from '@gooddata/sdk-model';
...
-import { IResultHeader } from '@gooddata/sdk-backend-spi';
-import { IResultMeasureHeader } from '@gooddata/sdk-backend-spi';
-import { IResultTotalHeader } from '@gooddata/sdk-backend-spi';
-import { IResultWarning } from '@gooddata/sdk-backend-spi';
+import { IResultAttributeHeader } from '@gooddata/sdk-model';
+import { IResultHeader } from '@gooddata/sdk-model';
+import { IResultMeasureHeader } from '@gooddata/sdk-model';
+import { IResultTotalHeader } from '@gooddata/sdk-model';
+import { IResultWarning } from '@gooddata/sdk-model';
 import { ISeparators } from '@gooddata/sdk-backend-spi';
```
Comment (no23reason) [inline libs/sdk-ui/api/sdk-ui.api.md:53]: Please check the ISeparators, they should now only come from sdk-model.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/7496/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/9356/ : SUCCESS in 16m 42s

File: libs/sdk-ui-dashboard/api/sdk-ui-dashboard.api.md (hunk 2 above)
Comment (kandl) [reply libs/sdk-ui-dashboard/api/sdk-ui-dashboard.api.md]: Removed.

File: libs/sdk-ui/api/sdk-ui.api.md (hunk 4 above)
Comment (kandl) [reply libs/sdk-ui/api/sdk-ui.api.md]: Removed.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/7502/ : SUCCESS in 10s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/9362/ : SUCCESS in 17m 33s

**Review by no23reason — APPROVED** [review]
LGTM

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR2415

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/2171/ : SUCCESS in 2m 49s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/2457/ : SUCCESS in 28s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/1280/ : SUCCESS in 20m 12s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/9363/ : SUCCESS in 20m 03s

**Pull request merged**